### PR TITLE
fix(availability): [#RBS-154] fix sync availabilities for user with many resources

### PR DESCRIPTION
**Pull request by qperie**
## Describe your changes
Call a unique API to get all the availabilty/unavaibability for the user and filter in front by resource instead of calling API to get them for resource by resource.

## Issue ticket number and link
RBS-154 : https://jira.support-ent.fr/browse/RBS-154

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

**Review by alicedraillard — APPROVED** [review]

File: src/main/java/net/atos/entng/rbs/controllers/AvailabilityController.java [hunk 1]
```diff
@@ -27,9 +27,11 @@
 import io.vertx.core.logging.Logger;
 import io.vertx.core.logging.LoggerFactory;
 import net.atos.entng.rbs.core.constants.Actions;
+import net.atos.entng.rbs.core.constants.Field;
 import net.atos.entng.rbs.models.Availability;
+import net.atos.entng.rbs.service.impl.AvailabilityServiceSqlImpl;
+import net.atos.entng.rbs.service.impl.ResourceServiceSqlImpl;
```
Comment (BastienBenard) [inline src/main/java/net/atos/entng/rbs/controllers/AvailabilityController.java:33]: import inutilisé

File: src/main/java/net/atos/entng/rbs/service/IcalExportService.java [hunk 2]
```diff
@@ -24,6 +24,7 @@
 import net.atos.entng.rbs.BookingUtils;
 import net.atos.entng.rbs.model.ExportBooking;
 import net.atos.entng.rbs.model.ExportResponse;
+import net.atos.entng.rbs.service.impl.UserServiceDirectoryImpl;
```
Comment (BastienBenard) [inline src/main/java/net/atos/entng/rbs/service/IcalExportService.java:27]: import inutilisé

File: src/main/java/net/atos/entng/rbs/service/impl/BookingServiceSqlImpl.java [hunk 3]
```diff
@@ -34,7 +34,7 @@
 import net.atos.entng.rbs.model.ExportBooking;
 import net.atos.entng.rbs.model.ExportRequest;
 import net.atos.entng.rbs.models.Slots;
-import org.entcore.common.bus.BusResponseHandler;
+import net.atos.entng.rbs.service.BookingService;
```
Comment (BastienBenard) [inline src/main/java/net/atos/entng/rbs/service/impl/BookingServiceSqlImpl.java:37]: import inutilisé

File: src/main/resources/public/ts/models/models.ts [hunk 4]
```diff
@@ -857,70 +858,87 @@ model.build = function () {
...
+                let allAvailabilities = new Availabilities();
+                let { data, status } = await availabilityService.list();
+                if (status == 200) {
+                    for (let a of data) {
+                        let tempAvailability = new Availability();
+                        tempAvailability.setFromJson(a);
+                        allAvailabilities.all.push(tempAvailability);
+                    }
+                }
+
```
Comment (BastienBenard) [inline src/main/resources/public/ts/models/models.ts:881]: typage si possible de allAvailabilities status data et a

**Review by BastienBenard — APPROVED** [review]

File: src/main/resources/public/ts/models/models.ts (hunk 4 above)
Comment (qperie) [reply src/main/resources/public/ts/models/models.ts:881]: C'est les prop de AxiosResponse, elles sont déjà typées any et number de façon natives

File: src/main/java/net/atos/entng/rbs/controllers/AvailabilityController.java (hunk 1 above)
Comment (qperie) [reply src/main/java/net/atos/entng/rbs/controllers/AvailabilityController.java:33]: Si, c'est pour les lignes 56 et 57 qui existaient déjà

**Pull request merged**